root/static/scripts/release-editor/dialogs.js

```diff
@@ -20,341 +22,359 @@ import trackParser from './trackParser';
-                    <bdi>{this.name}</bdi>
-                    {formatString ? ' ' + bracketed(formatString) : null}
-                </>
-            ),
-            artist: <bdi>{this.artist}</bdi>,
-        });
+    const link = exp.l('{entity} by {artist}', {
+      entity: (
+        <>
+          <bdi>{this.name}</bdi>
+          {formatString ? ' ' + bracketed(formatString) : null}
+        </>
+      ),
+      artist: <bdi>{this.artist}</bdi>,
+    });
 
-        return ReactDOMServer.renderToStaticMarkup(link);
-    }
+    return ReactDOMServer.renderToStaticMarkup(link);
+  }
 }
 
 
 class SearchTab {
-
-    constructor() {
-        this.releaseName = ko.observable("");
-        this.artistName = ko.observable("");
-        this.trackCount = ko.observable("");
-
-        this.searchResults = ko.observable(null);
-        this.result = ko.observable(null);
-        this.searching = ko.observable(false);
-        this.error = ko.observable("");
-
-        this.currentPage = ko.observable(0);
-        this.totalPages = ko.observable(0);
+  constructor() {
+    this.releaseName = ko.observable('');
+    this.artistName = ko.observable('');
+    this.trackCount = ko.observable('');
+
+    this.searchResults = ko.observable(null);
+    this.result = ko.observable(null);
+    this.searching = ko.observable(false);
+    this.error = ko.observable('');
+
+    this.currentPage = ko.observable(0);
+    this.totalPages = ko.observable(0);
+  }
+
+  search(data, event, pageJump) {
+    this.searching(true);
+
+    const data = {
+      q: this.releaseName(),
+      artist: this.artistName(),
+      tracks: this.trackCount(),
+      page: pageJump ? this.currentPage() + pageJump : 1,
+    };
+
+    this._jqXHR = request({url: this.endpoint, data: data}, this)
+      .done(this.requestDone)
+      .fail(function (jqXHR, textStatus) {
+        if (textStatus !== 'abort') {
+          this.error(jqXHR.responseText);
+        }
+      })
+      .always(function () {
+        this.searching(false);
+      });
+  }
+
+  cancelSearch() {
+    if (this._jqXHR) {
+      this._jqXHR.abort();
     }
-
-    search(data, event, pageJump) {
-        this.searching(true);
-
-        var data = {
-            q: this.releaseName(),
-            artist: this.artistName(),
-            tracks: this.trackCount(),
-            page: pageJump ? this.currentPage() + pageJump : 1
-        };
-
-        this._jqXHR = request({ url: this.endpoint, data: data }, this)
-            .done(this.requestDone)
-            .fail(function (jqXHR, textStatus) {
-                if (textStatus !== "abort") {
-                    this.error(jqXHR.responseText);
-                }
-            })
-            .always(function () {
-                this.searching(false);
-            });
+  }
+
+  buttonClicked() {
+    this.searching() ? this.cancelSearch() : this.search();
+  }
+
+  keydownEvent(data, event) {
+    if (event.keyCode === 13) { // Enter
+      this.search(data, event);
```

Comment:
consistent-return wants this to also return something. Should we, or should we just disable consistent-return for this bit?